hedera-node/src/main/java/com/hedera/services/utils/UnzipUtility.java

```diff
@@ -68,14 +68,17 @@ public static void unzip(byte[] bytes, String dstDirectory) throws IOException {
 		ZipEntry entry = zipIn.getNextEntry();
 		while (entry != null) {
 			String filePath = dstDirectory + File.separator + entry.getName();
+			File fileOrDir = new File(filePath);
+			filePath = fileOrDir.getCanonicalPath();
+			if(!filePath.startsWith(destDir.getCanonicalPath() + File.separator)) {
+				throw new IllegalArgumentException("Zip file entry " + filePath + " has potential security risk");
```

Comment:
Let's modify the comment so it doesn't say "security risk", but just says something like "the supplied path was invalid. Only specify a filePath with a full canonical path name" or something similar (you can come up with a better text).

Let's also make sure to write the tests including the negative tests to make sure this code works as expected and throws when needed.

Reply:
Re-worded the description and added test cases.